api/v1alpha1/common_types.go

```diff
@@ -345,6 +348,8 @@ type NodeState struct {
 	// LastUpdatedTime is the last time at which this node was updated. This must be sortable.
 	// +optional
 	LastUpdatedTime metav1.Time `json:"lastUpdatedTime,omitempty"`
+	// Certificate expire date contains the date when the certificate should be renewed
+	CertificateExpireDate CertificateExpireDate `json:"certificateExpireDate,omitempty"`
```

Comment:
This can be a `metav1.Time` like the `LastUpdatedTime` above it 🙂 

Then you don't need to do any time conversions. The cert-manager API returns a `Time` as well

https://github.com/cert-manager/cert-manager/blob/v1.7.2/pkg/apis/certmanager/v1/types_certificate.go#L410